src/lib/logs/log_integration_service.py

```diff
@@ -0,0 +1,117 @@
+import time
+import asyncio
+from typing import List, Tuple
+from lib.clientsession_provider import init_gcp_client_session, init_dt_client_session
+from lib.context import LoggingContext, LogsProcessingContext, create_logs_context
+from lib.credentials import create_token
+from lib.logs.dynatrace_client import DynatraceClient
+from lib.logs.gcp_client import GCPClient
+from lib.logs.log_self_monitoring import put_sfm_into_queue
+from lib.logs.logs_processor import (
+    prepare_context_and_process_message,
+    prepare_batches,
+    LogBatch,
+)
+from lib.utilities import chunks
+from lib.logs.log_forwarder_variables import (
+    REQUEST_BODY_MAX_SIZE,
+    NUMBER_OF_CONCURRENT_MESSAGE_PULL_COROUTINES,
+    NUMBER_OF_CONCURRENT_PUSH_COROUTINES,
+)
+
+
+class LogIntegrationService:
+    def __init__(self, sfm_queue: asyncio.Queue):
+        self.gcp_client = None
+        self.dynatrace_client = DynatraceClient()
+        self.sfm_queue = sfm_queue
+        self.log_push_semaphore = asyncio.Semaphore(NUMBER_OF_CONCURRENT_PUSH_COROUTINES)
+
+    async def update_gcp_client(self, logging_context: LoggingContext):
+        async with init_gcp_client_session() as gcp_session:
+            token = await create_token(logging_context, gcp_session)
+            if token is None:
+                raise Exception("Cannot start pub/sub pulling - Failed to fetch token")
```

Comment:
Is it possible to make this Exception more informative - like why we failed to fetch the token?